com.microsoft.mrtk.input/Controllers/ArticulatedHandController.cs

```diff
@@ -97,17 +95,27 @@ protected override void UpdateInput(XRControllerState controllerState)
                     // Workaround for missing select actions on devices without interaction profiles
                     // for hands, such as Varjo and Quest. Should be removed once we have universal
                     // hand interaction profile(s) across vendors.
+                    
+                    // Debounce the polyfill pinch action value.
+                    bool isPinched = pinchAmount >= (pinchedLastFrame ? 0.9f : 1.0f);
+
+                    // Inject our own polyfilled state into the Select state if no other control is bound.
                     if (!selectAction.action.HasAnyControls())
                     {
-                        // Debounced.
-                        bool isPinched = pinchAmount >= (pinchedLastFrame ? 0.9f : 1.0f);
-
                         controllerState.selectInteractionState.active = isPinched;
                         controllerState.selectInteractionState.activatedThisFrame = isPinched && !pinchedLastFrame;
                         controllerState.selectInteractionState.deactivatedThisFrame = !isPinched && pinchedLastFrame;
+                    }
 
-                        pinchedLastFrame = isPinched;
+                    // Also make sure we update the UI press state.
+                    if (!uiPressAction.action.HasAnyControls())
+                    {
+                        controllerState.uiPressInteractionState.active = isPinched;
```

Comment:
Do you also need to set `controllerState.uiPressInteractionState.value` like we do for `selectInteractionState` on line 93?

Reply:
Nothing currently uses `uiPress.value` in XRI but sure!

Also, moving line 93 down inside the `.HasAnyControls` check.